src/Analyzer.Utilities/Extensions/IEnumerableOfIMethodSymbolExtensions.cs

```diff
@@ -0,0 +1,92 @@
+                 int j = 0;
+
+                if (!trailingOnly && candidateMethod.Parameters.First().Type.Equals(expectedParameterType))
+                {
+                     // If expectedParameterType is the first parameter then the parameters to compare in candidateMethod against selectedOverload
+                     // is offset by 1
+                     j = 1;
+                }
+                else if (!candidateMethod.Parameters.Last().Type.Equals(expectedParameterType))
+                {
+                     // expectedParameterType is neither the first parameter nor the last parameter
+                     return false;
+                }
+
+                for (int i = 0; i < selectedOverload.Parameters.Count(); i++, j++)
+                {
+                    if (!selectedOverload.Parameters[i].Type.Equals(candidateMethod.Parameters[j].Type) ||
+                        selectedOverload.Parameters[i].IsParams != candidateMethod.Parameters[j].IsParams)
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
+            });
+        }
+
+        /// <summary>
+        /// Returns a list of method symbols from a given list of the method symbols, which has its parameter type as
+        /// expectedParameterType as its last parameter in addition to matching all the other parameter types of the 
+        /// selectedOverload method symbol
+        /// </summary>
+        /// <param name="methods">List of <see cref="IMethodSymbol"/> to scan for possible overloads</param>
+        /// <param name="selectedOverload"><see cref="IMethodSymbol"/> that is currently picked by the user</param>
+        /// <param name="expectedParameterType"><see cref="INamedTypeSymbol"/> type of the leading parameter or the trailing parameter</param>
+        public static IEnumerable<IMethodSymbol> GetMethodOverloadsWithDesiredParameterAtTrailing(
+             this IEnumerable<IMethodSymbol> methods,
+             IMethodSymbol selectedOverload,
+             INamedTypeSymbol expectedParameterType)
+        {
+            return GetMethodOverloadsWithDesiredParameterAtLeadingOrTrailing(methods, selectedOverload, expectedParameterType, true);
+        }
+
+        /// <summary>
+        /// Return all the method symbols with the name matching the displayName case-insensitively
+        /// </summary>
+        /// <param name="stringFormatMembers"></param>
+        /// <param name="displayName"></param>
+        /// <returns></returns>
+        public static IMethodSymbol GetSingleOrDefaultMemberWithName(this IEnumerable<IMethodSymbol> stringFormatMembers, string displayName)
+        {
+             return stringFormatMembers?.Where(member => string.Equals(member.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat), displayName, StringComparison.OrdinalIgnoreCase)).SingleOrDefault();
```

Comment:
This uses a language specific format, please move this helper to the analyzer file.